native/src/main/java/io/ballerina/lib/data/xmldata/utils/DataUtils.java

```diff
@@ -465,10 +468,23 @@ public static Object getModifiedRecord(BMap<BString, Object> input, BString text
         return input;
     }
 
+    private static BMap<BString, Object> mergeOriginalAndCurrentAnnotations(
+                                BMap<BString, Object> originalTypeAnnotations, BMap<BString, Object> typeAnnotations) {
+        BMap<BString, Object> annotations = originalTypeAnnotations;
+        for (Map.Entry<BString, Object> entry: typeAnnotations.entrySet()) {
+            if (annotations.containsKey(entry.getKey())) {
+                continue;
+            }
+            annotations.put(entry.getKey(), entry.getValue());
+        }
+        return annotations;
+    }
```

Comment:
```java
private static BMap<BString, Object> mergeOriginalAndCurrentAnnotations(
        BMap<BString, Object> originalTypeAnnotations, BMap<BString, Object> typeAnnotations) {

    BMap<BString, Object> mergedAnnotations = BMap.create(originalTypeAnnotations); // Ensure a new modifiable map

    for (Map.Entry<BString, Object> entry : typeAnnotations.entrySet()) {
        mergedAnnotations.putIfAbsent(entry.getKey(), entry.getValue());
    }
    
    return mergedAnnotations;
}
```

Reply:
Why we neesd to create a new map in here?

Reply:
We can keep the original map unchanged. If we changed that, it can effect.